pymc4/distributions/mixture.py

```diff
@@ -19,11 +26,36 @@ class Mixture(Distribution):
     ----------
     p : array of floats
         p >= 0 and p <= 1
-        the mixture weights, in the form of probabilities
+        the mixture weights, in the form of probabilities, must sum to one.
     distributions : multidimensional PyMC4 distribution (e.g. `pm.Poisson(...)`)
         or iterable of one-dimensional PyMC4 distributions the
         component distributions :math:`f_1, \ldots, f_n`
     """
 
     def __init__(self, name, p, distributions, **kwargs):
         super().__init__(name, p=p, distributions=distributions, **kwargs)
+
+    @staticmethod
+    def _init_distribution(conditions, **kwargs):
+        p, d = conditions["p"], conditions["distributions"]
+        # if 'd' is a list of pymc distributions, then use the underlying
+        # tfp distributions for the mixture
+        if isinstance(d, list):
+            distributions = [el._distribution for el in d]
+        # if 'd' is a pymc distribution with batch_size > 1, then build
+        # K tfp distributions for tfd.mixture
+        else:
+            # get class of tfd distribution, i.e. Normal/Poisson/etc.
+            ty = type(d._distribution)
```

Comment:
I don't agree with this branch of the if statement. We should
1. Test that `d` is a `pm.Distribution`
2. If the test passes, feed it into `MixtureSameFamily`
3. Raise a `TypeError` otherwise

No intermediate tfd.Distribution other than the mixture and the Categorical should be created in this function.

Reply:
Oh, that is way more elegant. However, if only `Categorical` and `Mixture` should are created, that means that a list of distributions of the same type, say `[pm.Poisson(), pm.Poisson(), pn.Poisson()` shouldn't be coalesced to a single Poisson which would allow to use `MistureSameFamily`, right?

Reply:
Yes, if the user feeds in a list of distributions, we treat them as if they weren't of the same family. We only defer to the `MixtureSameFamily` if the user supplies a distribution instance.